compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -910,10 +901,23 @@ struct ROCMSession final
 
 } // namespace mlir::iree_compiler::IREE::HAL
 
+// Iterate over ukernel bitcode embedded-data files, and insert them into the
+// EmbeddedDataDirectory singleton.
+static void AddAMDGPUUkernelBitcodeToGlobalEmbeddedDataDirectory() {
+  using mlir::iree_compiler::EmbeddedDataDirectory;
+  std::lock_guard<std::mutex> lock(EmbeddedDataDirectory::getGlobalMutex());
+  auto &dir = EmbeddedDataDirectory::getGlobal();
+  const iree_file_toc_t *toc = iree_uk_amdgpu_bitcode_create();
+  for (size_t i = 0; i < iree_uk_amdgpu_bitcode_size(); ++i) {
```

Comment:
nit: don't reevaluate size after every iteration

Reply:
`iree_uk_amdgpu_bitcode_size()` is inline in the generated header file:
```c++
static inline size_t iree_uk_amdgpu_bitcode_size() {
  return 16;
}
```